db/inst/import-try/04.match.species.R

```diff
@@ -0,0 +1,96 @@
+# 4. Match species between TRY and BETY
+source("common.R")
+load("try.3.RData")
+
+# Get unique species list from BETY
+# NOTE: This is set to PSQL-PEcAn because my clone of BETY was missing all the species
+dbparms.psql <- list(driver="PostgreSQL" , user = "bety", dbname = "bety", password='bety', host='psql-pecan.bu.edu')
+con.psql <- db.open(dbparms.psql)
+bety.species.dt <- data.table(db.query("SELECT DISTINCT id as bety_species_id, scientificname as bety_species FROM species", con.psql))
+bety.species.dt[, bety.species.lower := tolower(bety_species)]
+db.close(con.psql)
+
+# a. Get unique species list from TRY
+try.species.unique <- try.dat[, unique(AccSpeciesName)]
+try.species.dt <- data.table(try.species = try.species.unique,
+                             try.species.lower = tolower(encodeString(try.species.unique)))
+try.species.dt[, try.species.lower := gsub(" sp$", " spp.", try.species.lower)]
+
+
+# Match TRY and BETY species.
+## First a literal match (because it's fast in data.table)
+setkey(try.species.dt, try.species.lower)
+setkey(bety.species.dt, bety.species.lower)
+match.species.dt <- bety.species.dt[try.species.dt]
+try.unmatched <- match.species.dt[is.na(bety_species_id)]
+
+## Second, a partial pattern match using grep
+# TODO: Use taxize gnr_resolve() instead of grep for matching.
```

Comment:
@dlebauer This is where the `taxize` functions would go. If we get a more recent version of R, this code can be modified to use that -- it'll definitely work better than the direct matching used here, which probably missed a lot of duplicates.